src/mimeparser.rs

```diff
@@ -1302,6 +1303,18 @@ fn get_attachment_filename(mail: &mailparse::ParsedMail) -> Result<Option<String
     let desired_filename =
         desired_filename.or_else(|| mail.ctype.params.get("name").map(|s| s.to_string()));
 
+    // decode filename as CHARSET'LANG'test%2E%70%64%66.
+    // we're always assuming utf-8,
```

Comment:
There is a separate issue for handling non-utf8 filenames: #1510
So it is really used and needs a fix, I have added it to the project.

Reply:
i just noticed also the iteration by just concatenating all `filename` keys is not correct. eg. the "Kopano" mailing webapp writes the filename twice, one time word-encoded and a second time with this apostrophe-encoding:
```
Content-Disposition: attachment; filename="=?utf-8?Q?Ma=C3=9Fnahmen_Okt=2E_2020=2Epdf?=";
 filename*=utf-8''Ma%C3%9Fnahmen%20Okt.%202020.pdf
```